app/src/main/java/ar/scacchipa/twittercloneapp/domain/RevokeCredentialsUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package ar.scacchipa.twittercloneapp.domain
+
+import android.content.SharedPreferences
+import ar.scacchipa.twittercloneapp.data.ResponseDomain
+import ar.scacchipa.twittercloneapp.data.RevokeDomain
+import ar.scacchipa.twittercloneapp.repository.IRevokeTokenRepository
+import ar.scacchipa.twittercloneapp.utils.Constants
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class RevokeCredentialsUseCase(
+    private val repository: IRevokeTokenRepository,
```

Comment:
Same here you can have as manny repositories depencies as you want in your use case, but you can not have a data source.